### DISCUSSION TITLE
Build IGT

### DISCUSSION THREAD

**Discussion by Nega100** [jspsych/jsPsych#1309] [Q&A]
Hi, Josh
I trying to build a IGT paradigm using JsPsych. Using html-keyboard-response, I successfully make the deck card presented. However the logic math behind the card reward and punishment are still confusing. Anyone here could help me with just a little clue what I am supposed to do?. Thanks a lot and very appreciate for the help. I would share the code when is finished.

**Comment by jodeleeuw** [comment]
Hi @Nega100 ,

This is a pretty broad question. Can you narrow it down a bit? What aspect of the task are you stuck with? If you can ask separate focused questions it will be easier to help.

**Reply by Nega100** [reply]
Hi Josh,
Really sorry for the delay. For a moment I read and gather more information from this forum discussion about building this paradigm. However I'm stuck and dont go anywhere. I'm really apreciated for your and community help.
`var totalmoney = 2000,
      slotA = 0, 
      slotB = 0,
      slotC = 0,
      slotD = 0,
      totalchoose = 0, 
      penalty = 0,   
      scoregain = 0,   
      slotA_WIN = 100,
      slotB_WIN = 100,
      slotC_WIN = 50,
      slotD_WIN = 50,
      maxmoney = 10000,
      maxplay = 100;
      
  var slotApenalty = [0, 0, -150, 0, -300, 0, -200, 0, -250, -350, 0, -350, 0, -250, -200, 0, -300, -150, 0, 0, 0, -300, 0, -350, 0, -200, -250, -150, 0, 0, -350, -200, -250, 0, 0, 0, -150, -300, 0, 0];
  var slotBpenalty = [0, 0, 0, 0, 0, 0, 0, 0, -1250, 0, 0, 0, 0, -1250, 0, 0, 0, 0, 0, 0, -1250, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1250, 0, 0, 0, 0, 0, 0, 0, 0];
  var slotCpenalty = [0, 0, -50, 0, -50, 0, -50, 0, -50, -50, 0, -25, -75, 0, 0, 0, -25, -75, 0, -50, 0, 0, 0, -50, -25, -50, 0, 0, -75, -50, 0, 0, 0, -25, -25, 0, -75, 0, -50, -75];
  var slotDpenalty = [0, 0, 0, 0, 0, 0, 0, 0, 0, -250, 0, 0, 0, 0, 0, 0, 0, 0, 0, -250, 0, 0, 0, 0, 0, 0, 0, 0, -250, 0, 0, 0, 0, 0, -250, 0, 0, 0, 0, 0];
  var selectedCards = []; 

  var trial_1 = {
    type: 'html-button-response',
    stimulus: " ",
    choices: ['<img id = "decka" src="IGT/examples/img/deck_a.png"></img>',
                 '<img id = "deckb" src="IGT/examples/img/deck_b.png"></img>',
                 '<img id = "deckc" src="IGT/examples/img/deck_c.png"></img>',
                 '<img id = "deckd" src="IGT/examples/img/deck_a.png"></img>'],
    maintain_aspect_ratio: true,
    post_trial_gap: 500,
    prompt: " ",
    on_finish : function(data){
      var responses = JSON.parse(data.responses);
      case "decka":
        if (slotA === slotApenalty.length){
          slotA = 0;
          }   
          penalty = slotApenalty[slotA];
          scoregain = slotA_WIN + penalty;                                              
          deckAclicks++;                       
          selectedCards.push("A");                                
          break;
        }
  }`

**Comment by gorkang** [comment]
@Nega100 , we implemented a version of the IGT (see attachment: [iowa_gambling_task.zip](https://github.com/jspsych/jsPsych/files/5751396/iowa_gambling_task.zip))

It is in Spanish but should be easy enough to adapt and improve.

Hopefully it can be helpful to you.

**Reply by Nega100** [reply]
Hi, @gorkang you a life saver!!. Thankss for your generosity to sharing your code. It really help me a lot and maybe other people who want to use this paradigm with jspsych for their research. Thanks again and this great community!.
